lib/pdf_fill/extras_generator_v2.rb

```diff
@@ -5,6 +5,68 @@ class ExtrasGeneratorV2 < ExtrasGenerator
     HEADER_FONT_SIZE = 14.5
     SUBHEADER_FONT_SIZE = 10.5
 
+    class Question
+      attr_accessor :section_index, :overflow
+
+      def initialize(metadata)
+        @section_index = nil
+        @number = metadata[:question_num]
+        @subquestions = []
+        @overflow = false
+      end
+
+      def add_text(value, metadata)
+        @subquestions << { value:, metadata: }
+        @overflow ||= metadata.fetch(:overflow, true)
+      end
+
+      def sorted_subquestions
+        @subquestions.sort_by { |subq| [subq[:metadata][:question_suffix], subq[:metadata][:question_text]] }
```

Comment:
If :question_suffix or :question_text is nil, this will raise an error.

You probably want to update this to

```
@subquestions.sort_by do |subq|
  suffix = subq[:metadata][:question_suffix] || ''
  text   = subq[:metadata][:question_text] || ''
  [suffix, text]
end
```

Reply:
Great catch! I thought I could get away with it but mixing nils and strings does crash it. I added a new RSpec for this scenario.